src/env/tock/crypto.rs

```diff
@@ -12,148 +12,12 @@
-    }
-}
-
-pub struct TockEcdhSharedSecret {
-    shared_secret: [u8; EC_FIELD_BYTE_SIZE],
-}
-
-impl ecdh::SharedSecret for TockEcdhSharedSecret {
-    fn raw_secret_bytes(&self) -> [u8; EC_FIELD_BYTE_SIZE] {
-        self.shared_secret
-    }
-}
-
-impl ecdsa::Ecdsa for TockEcdsa {
-    type SecretKey = TockEcdsaSecretKey;
-    type PublicKey = TockEcdsaPublicKey;
-    type Signature = TockEcdsaSignature;
-}
-
-pub struct TockEcdsaSecretKey {
-    sec_key: crypto::ecdsa::SecKey,
-}
-
-impl ecdsa::SecretKey for TockEcdsaSecretKey {
-    type PublicKey = TockEcdsaPublicKey;
-    type Signature = TockEcdsaSignature;
-
-    fn random(rng: &mut impl Rng256) -> Self {
-        let sec_key = crypto::ecdsa::SecKey::gensk(rng);
-        Self { sec_key }
-    }
-
-    fn from_slice(bytes: &[u8; EC_FIELD_BYTE_SIZE]) -> Option<Self> {
-        crypto::ecdsa::SecKey::from_bytes(bytes).map(|k| Self { sec_key: k })
-    }
-
-    fn public_key(&self) -> Self::PublicKey {
-        let pub_key = self.sec_key.genpk();
-        TockEcdsaPublicKey { pub_key }
-    }
-
-    fn sign(&self, message: &[u8]) -> Self::Signature {
-        let signature = self.sec_key.sign_rfc6979::<crypto::sha256::Sha256>(message);
-        TockEcdsaSignature { signature }
-    }
-
-    fn to_slice(&self, bytes: &mut [u8; EC_FIELD_BYTE_SIZE]) {
-        self.sec_key.to_bytes(bytes);
-    }
-}
-
-pub struct TockEcdsaPublicKey {
-    pub_key: crypto::ecdsa::PubKey,
-}
-
-impl ecdsa::PublicKey for TockEcdsaPublicKey {
-    type Signature = TockEcdsaSignature;
-
-    fn from_coordinates(
-        x: &[u8; EC_FIELD_BYTE_SIZE],
-        y: &[u8; EC_FIELD_BYTE_SIZE],
-    ) -> Option<Self> {
-        crypto::ecdsa::PubKey::from_coordinates(x, y).map(|k| Self { pub_key: k })
-    }
-
-    fn verify(&self, message: &[u8], signature: &Self::Signature) -> bool {
-        self.pub_key
-            .verify_vartime::<crypto::sha256::Sha256>(message, &signature.signature)
-    }
-
-    fn to_coordinates(&self, x: &mut [u8; EC_FIELD_BYTE_SIZE], y: &mut [u8; EC_FIELD_BYTE_SIZE]) {
-        self.pub_key.to_coordinates(x, y);
-    }
-}
-
-pub struct TockEcdsaSignature {
-    signature: crypto::ecdsa::Signature,
-}
-
-impl ecdsa::Signature for TockEcdsaSignature {
-    fn from_slice(bytes: &[u8; EC_SIGNATURE_SIZE]) -> Option<Self> {
-        crypto::ecdsa::Signature::from_bytes(bytes).map(|s| TockEcdsaSignature { signature: s })
-    }
-
-    fn to_der(&self) -> Vec<u8> {
-        self.signature.to_asn1_der()
-    }
-}
+use opensk::api::crypto::software_crypto;
+
+pub use software_crypto::{
+    SoftwareCrypto as TockCrypto, SoftwareEcdh as TockEcdh,
+    SoftwareEcdhPublicKey as TockEcdhPublicKey, SoftwareEcdhSecretKey as TockEcdhSecretKey,
+    SoftwareEcdhSharedSecret as TockEcdhSharedSecret, SoftwareEcdsa as TockEcdsa,
+    SoftwareEcdsaPublicKey as TockEcdsaPublicKey, SoftwareEcdsaSecretKey as TockEcdsaSecretKey,
+    SoftwareEcdsaSignature as TockEcdsaSignature,
+};
```

Comment:
Same question here. Why not use directly `SoftwareFoo` syntax in the `TockEnv` implementation?

Reply:
I removed the file.